nat_traversal.nimble

```diff
@@ -10,21 +10,31 @@ installDirs   = @["vendor"]
 ### Dependencies
 requires "nim >= 0.19.0", "stew"
 
-proc compileStaticLibraries() =
+type Arch = enum
+  host, armv7
+
+proc compileStaticLibraries(arch: static Arch) =
   withDir "vendor/miniupnp/miniupnpc":
     when defined(windows):
       exec("mingw32-make -f Makefile.mingw CC=gcc init libminiupnpc.a")
+    elif arch == armv7:
+      exec("make -B CC=arm-linux-gnueabihf-gcc libminiupnpc.a")
     else:
-      exec("make libminiupnpc.a")
+      exec("make -B libminiupnpc.a")
   withDir "vendor/libnatpmp":
     when defined(windows):
       exec("mingw32-make CC=gcc CFLAGS=\"-Wall -Os -DWIN32 -DNATPMP_STATICLIB -DENABLE_STRNATPMPERR -DNATPMP_MAX_RETRIES=4\" libnatpmp.a")
+    elif arch == armv7:
+      exec("make -B CC=arm-linux-gnueabihf-gcc CFLAGS=\"-Wall -Os -DENABLE_STRNATPMPERR -DNATPMP_MAX_RETRIES=4\" libnatpmp.a")
     else:
-      exec("make CFLAGS=\"-Wall -Os -DENABLE_STRNATPMPERR -DNATPMP_MAX_RETRIES=4\" libnatpmp.a")
+      exec("make -B CFLAGS=\"-Wall -Os -DENABLE_STRNATPMPERR -DNATPMP_MAX_RETRIES=4\" libnatpmp.a")
```

Comment:
Wouldn't it be easier to do a `make clean` in there instead of forcing recompilation of these libraries, all the time, on non-Windows platforms?

Reply:
Yeah, that was the second option I considered. I thought recompilation was only done when all nim-beacon-chain targets are built (and `make update`) but I guess that's not the case

Reply:
Similar to this? https://github.com/status-im/nim-beacon-chain/blob/master/Makefile#L171

Reply:
Yes.